sdk-extensions/autoconfigure/src/test/java/io/opentelemetry/sdk/autoconfigure/OtlpConfigUtilTest.java

```diff
@@ -58,4 +67,121 @@ void getOtlpProtocolDefault() {
+        .doesNotThrowAnyException();
+  }
+
+  @Test
+  void configureOtlpExporterBuilder_InvalidEndpoints() {
+    assertThatThrownBy(
+            configureEndpoint(ImmutableMap.of("otel.exporter.otlp.endpoint", "/foo/bar")))
+        .isInstanceOf(ConfigurationException.class)
+        .hasMessageContaining("OTLP endpoint must be a valid URL:");
+    assertThatThrownBy(
+            configureEndpoint(
+                ImmutableMap.of("otel.exporter.otlp.endpoint", "file://localhost:4317")))
+        .isInstanceOf(ConfigurationException.class)
+        .hasMessageContaining("OTLP endpoint scheme must be http or https:");
+    assertThatThrownBy(
+            configureEndpoint(
+                ImmutableMap.of("otel.exporter.otlp.endpoint", "http://localhost:4317?foo=bar")))
+        .isInstanceOf(ConfigurationException.class)
+        .hasMessageContaining("OTLP endpoint must not have a query string:");
+    assertThatThrownBy(
+            configureEndpoint(
+                ImmutableMap.of("otel.exporter.otlp.endpoint", "http://localhost:4317#fragment")))
+        .isInstanceOf(ConfigurationException.class)
+        .hasMessageContaining("OTLP endpoint must not have a fragment:");
+    assertThatThrownBy(
+            configureEndpoint(
+                ImmutableMap.of(
+                    "otel.exporter.otlp.endpoint", "http://localhost:4317/path",
+                    "otel.exporter.otlp.protocol", "grpc")))
+        .isInstanceOf(ConfigurationException.class)
+        .hasMessageContaining("OTLP endpoint must not have a path:");
+  }
+
+  private static ThrowingCallable configureEndpoint(Map<String, String> properties) {
+    return () ->
+        OtlpConfigUtil.configureOtlpExporterBuilder(
+            DATA_TYPE_TRACES,
+            DefaultConfigProperties.createForTest(properties),
+            value -> {},
+            (value1, value2) -> {},
+            value -> {},
+            value -> {},
+            value -> {});
+  }
+
+  @Test
+  void configureOtlpExporterBuilder_HttpProtobufEndpoint() {
+    assertThat(configureHttpProtobufEndpoint(DATA_TYPE_TRACES, "http://localhost:4317"))
+        .isEqualTo("http://localhost:4317/v1/traces");
+    assertThat(configureHttpProtobufEndpoint(DATA_TYPE_TRACES, "http://localhost:4317/"))
+        .isEqualTo("http://localhost:4317/v1/traces");
+    assertThat(configureHttpProtobufEndpoint(DATA_TYPE_TRACES, "http://localhost:4317/foo"))
+        .isEqualTo("http://localhost:4317/foo/v1/traces");
+    assertThat(configureHttpProtobufEndpoint(DATA_TYPE_TRACES, "http://localhost:4317/foo/"))
+        .isEqualTo("http://localhost:4317/foo/v1/traces");
+    assertThat(configureHttpProtobufEndpoint(DATA_TYPE_TRACES, "http://localhost:4317/v1/traces"))
+        .isEqualTo("http://localhost:4317/v1/traces/v1/traces");
+    assertThat(configureHttpProtobufEndpoint(DATA_TYPE_TRACES, "http://localhost:4317/v1/metrics"))
+        .isEqualTo("http://localhost:4317/v1/metrics/v1/traces");
```

Comment:
this one makes sense to me (I mean...it's almost certainly wrong, but it's what the user requested). However, the one where we add `/v1/traces` when it's already on the end seems surprising to me. Yes, it's precisely what the user is requesting, but I suspect this will happen and users will complain. Are we were we don't want to check to make sure that the generic endpoint hasn't already had the `/v1/<signal>` tacked onto it before duplicating it?

Reply:
This is exactly what the spec is forbidding. Better the users notice the error earlier than much later when they have deployed that config everywhere and want to start using metrics and it ends up at `/v1/traces/v1/metrics`.
If you suggest that the configuration should be "smart" and detect any signal path (e.g., replacing `/v1/metrics` with `/v1/traces` for traces, leaving it in-place for metrics) I suggest a spec PR. I wouldn't be against it, if we can manage to log some warning at least...

Reply:
Maybe I misunderstood and you meant the signal-specific thing? That seems like a bug indeed. I'll comment above.

Reply:
No, I was suggesting that if the user provides a trace endpoint as the "generic" endpoint, that we don't mess with it for traces, since it's likely(?) that they are only using tracing, and don't care about metrics and what endpoint might have been populated for it.

Reply:
That's currently against the spec and I think delaying the error until the point in time where the user uses any other signal is not good. It would be better to ensure that the (404?) error that occurs with the doubled signal path is easy to find (e.g. by logging).
I could be convinced of the "fully smart" method, stripping off any known trailing signal path from the generic URL, but making *only* the configured path work seems to be a bad balance.

Reply:
ok. I understand that side of the argument. This is potentially a breaking change for some users, so we should be very sure to call it out very clearly in our release notes/CHANGELOG.  @jack-berg Can you craft a CHANGELOG entry for this PR? Thanks!